src/Listener/PodeRequest.cs

```diff
@@ -229,6 +228,29 @@ private bool ValidateCertificateCallback(object sender, X509Certificate certific
             return true;
         }
 
+        /// <summary>
+        /// Provides access to a buffer. The buffer is allocated only when first requested,
+        /// saving memory if it is never needed.
+        /// </summary>
+        protected virtual byte[] Buffer
+        {
+            get
+            {
+                if (_buffer == null)
+                {
+                    _buffer = new byte[BufferSize];
+                }
+                return _buffer;
+            }
+        }
+
+        /// <summary>
+        /// This virtual method returns a buffer to be used during the receive operation.
+        /// Derived classes can override this to customize buffer allocation behavior.
+        /// By default, it returns the lazily allocated Buffer property.
+        /// </summary>
+        protected virtual byte[] GetBuffer() => Buffer;
```

Comment:
Any specific reason to have this method, and not just override the `Buffer` property in PodeSignalRequest with the following instead?

```c#
protected override byte[] Buffer
{
    get => new byte[BufferSize];
}
```

Reply:
But the `Buffer` property is marked as `virtual` 👀

Reply:
This was my main question, why use the method and not the property? There shouldn't be any issues, and I've done similar in other places with virtual properties; the method feels a tad redundant 🤔